model/container_task_queue.go

```diff
@@ -0,0 +1,150 @@
+package model
+
+import (
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/model/task"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/message"
+	"github.com/pkg/errors"
+)
+
+// ContainerTaskQueue represents an iterator that represents an ordered queue of
+// container tasks that are ready be allocated a container.
+type ContainerTaskQueue struct {
+	queue    []task.Task
+	position int
+}
+
+// Next returns the next task that's ready for container allocation. It will
+// return a nil task once there are no tasks remaining in the queue.
+func (q *ContainerTaskQueue) Next() (*task.Task, error) {
+	if q.queue == nil {
+		if err := q.populate(); err != nil {
+			return nil, errors.Wrap(err, "initial population of container task queue")
+		}
+	}
+	if q.position >= len(q.queue) {
+		return nil, nil
+	}
+
+	next := q.queue[q.position]
+
+	q.position++
```

Comment:
Do we not want to pop ever?

Reply:
For now, there's no need to have any operation other than pop. This is only used to cron populate pod allocator jobs.

Reply:
I meant, is it worth it to delete the element from the slice?

Reply:
I don't think so. There's no cost/benefit to deleting elements for now since it's in-memory. There might have to be some deletion/shuffling anyways once task groups are considered.